docs/specifications/4-standard-compliance-suite.md

```diff
@@ -0,0 +1,166 @@
+  test case, for which the release included a revision of the test case. Trivial
+  revisions (such as correcting typos, or re-phrasing the description without
+  changing the tests case's semantic value) should not be included in the list.
+
+After the attributes table, a **TypeScript** block of code describes the
+canonical form of the test. It includes any type declaration that is used by the
+test (so the code example is self-contained). Assertions performed by the test
+should be written in the form of [`jest`] expectations.
+
+[`jest`]: https://jestjs.io/docs/en/getting-started
+
+Finally, another code block details the sequence of messages that should be
+exchanged between the *host* and `node` processes during the execution of the
+test case, such that implementations can assert conherent behavior. The dialogue
+is the sequence of JSON formatted messages, from the perspective of the *host*
+app, using the following notation:
+* Messages sent by the *host* runtime to the `node` process:
+  ```
+  > { "api": "foo" }
+  ```
+* Messages received by the *host* runtime from the `node` process:
+  ```
+  < { "result": "bar" }
+  ```
+* Comments to improve readability of the trace:
+  ```
+  # Comment continues until the end of the line
+  ```
+* Blank lines can be added to logically group trace elements
+
+> :question: is there a need to support some form of a capture mechanism to
+> provision for non-deterministric results, or non-normative elements such as
+> the exact Object IDs issued for created instances?
+
+#### Template
+
+Below is the template markdown to copy-paste when introducing a new test case in
+the compliance suite. New tests should always be added at the very end of the
+category they belong to, right after the last test in said category.
+
+````md
+### Test Category
+#### Test Case Name
+
+A short english language description of what property this test verifies. The
+description should include enough detail for a reader to be able to understand
+the test without having to search for any additional information. Prefer a long,
+unambiguous description to a terse one that could be subject to interpretation.
+
+Attribute      | Value
+---------------|----------------------------------------------------------------
+Introduced     | 1.2.3
+Related Issues | #123, aws/aws-cdk#456
+
+##### Reference Implementation
+```ts
+// Declare any type used by the test
+export class Foo { /* ... */ }
+
+// Instructions the test will execute, using `jest`-like expectation syntax
+expect(new Foo().bar().baz)
+  .toBeUndefined();
+```
+
+##### Reference Kernel Messaging
+```
+< { "hello": "@jsii/runtime@1.2.3" }
+> { "load": { "name": "test-case-001", "version": "1.2.3", "tarball": "/tmp/jsii-kernel-test/lib.tgz" } }
+< { "assembly": "test-case-001", "types": 3 }
+```
+````
+
+## Compliance
+In order to be able to assert compliance of language binding libraries to the
```

Comment:
I like the idea of a standardized report, but why not make it some flavor of XUnit reporting XML? That has been semi-standardized and can be generated by nearly every unit testing tool easily enough.

Obviates the need for another XUnit -> JSON conversion tool.

Reply:
I mean if there is a standard format that satisfies the requirements I'm all for it. XML is meh to me, but if there's an XML standard, that wins. I'll investigate.